src/biotite/structure/bonds.pyx

```diff
@@ -1646,6 +1648,38 @@ def connect_via_residue_names(atoms, bint inter_residue=True,
             )
 
         atom_names_in_res = atom_names[curr_start_i : next_start_i]
+
+        # Check if we should use alternative atom names
+        std_atom_ids = get_from_ccd(
+            "chem_comp_atom", 
+            res_names[curr_start_i], 
+            "atom_id"
+        ) 
+        if (atom_names_in_res is not None and \
+            std_atom_ids is not None and \
+            not set(atom_names_in_res).issubset(std_atom_ids)):
+            # We do not assume that the order of atoms within 
+            # atom_names_in_res matches that of the CCD
+            alt_atom_ids = get_from_ccd(
+                "chem_comp_atom", 
+                res_names[curr_start_i], 
+                "alt_atom_id"
+            ) 
+            if set(atom_names_in_res).issubset(alt_atom_ids):
+                # Standardize atom IDs
+                mapping = dict(zip(alt_atom_ids, std_atom_ids))
+                mapped_atom_names_in_res = np.vectorize(
+                    mapping.get
+                )(atom_names_in_res)
```

Comment:
I believe as written we only call for residues that include non-standard atom names, not every residue (since we're looping over atoms within `atom_names_in_res`); the use of `vectorize` was only for convenience. Happy to switch to the equivalent loop if preferred for readability.

Reply:
You are right, since we now run it for residues with non-standard atoms, performance should be a non-issue. I still slightly prefer a list comprehension for readability, but I have no strong opinion here, so you can choose.